arm64 architecture support

### What problem are you facing?

We are trying to switch the architecture of some machines from `amd64` to `arm64` and found  that function pod starts _CrashLoopBackOff_-ing as soon as there's a single `XR`/`CompositeResource` referencing it. We switched back to an amd node and it works properly again. Could it be that arm64 arch is unsupported?

Info:
- Function version: `v0.7.2`
- Currently using AWS EKS:`v1.28.8-eks-ae9a62a`
- Nodes:
  - ARM64: AWS `m6g.2xlarge` instance
  - AMD64: AWS `m7i-flex.2xlarge` instance (Everything ok here)

```sh
crossplane beta trace -o wide <xrd>/<claim>
# NAME               RESOURCE   SYNCED   READY   STATUS
# <xrd>/<claim>                 True     False   Waiting: Claim is waiting for composite resource to become Ready
# └─ <xrd>/<xr>                 False    -       ReconcileError: cannot compose resources: cannot run Composition pipeline step "kcl-render": cannot run Function "function-kcl": rpc error: code = Unavailable desc = last connection error: connection error: desc = "transport: Error while dialing: dial tcp x.x.x.x:9443: connect: connection refused"
```

<details>
  <summary>KCL Function Pod - Describe</summary>

```sh
kubectl describe po -l pkg.crossplane.io/function=function-kcl
```

```
Name:             function-kcl-c44536185222-66c9f554fd-m8wht
Namespace:        crossplane
Priority:         0
Service Account:  function-kcl-c44536185222
Node:             ...
Start Time:       Fri, 10 May 2024 11:21:05 +0200
Labels:           pkg.crossplane.io/function=function-kcl
                  pkg.crossplane.io/revision=function-kcl-c44536185222
                  pod-template-hash=66c9f554fd
Annotations:      vpc.amazonaws.com/pod-ips: x.x.x.x
Status:           Running
IP:               x.x.x.x
IPs:
  IP:           x.x.x.x
Controlled By:  ReplicaSet/function-kcl-c44536185222-66c9f554fd
Containers:
  package-runtime:
    Container ID:   containerd://031093bf8ba71fa1ff3daa11ffd3df236d605794365adc4753851789f044d2d8
    Image:          xpkg.upbound.io/crossplane-contrib/function-kcl:v0.7.2
    Image ID:       xpkg.upbound.io/crossplane-contrib/function-kcl@sha256:c4453618522289c708d15cb12cd0f35ba0c612d1c6c4d7dcd095aa12ef97d4e5
    Ports:          8080/TCP, 9443/TCP
    Host Ports:     0/TCP, 0/TCP
    State:          Running
      Started:      Fri, 10 May 2024 11:23:16 +0200
    Last State:     Terminated
      Reason:       Error
      Exit Code:    2
      Started:      Fri, 10 May 2024 11:23:01 +0200
      Finished:     Fri, 10 May 2024 11:23:01 +0200
    Ready:          True
    Restart Count:  2
    Environment:
      TLS_SERVER_CERTS_DIR:             /tls/server
    Mounts:
      /tls/server from tls-server-certs (ro)
      /var/run/secrets/kubernetes.io/serviceaccount from kube-api-access-4q9sf (ro)
Conditions:
  Type              Status
  Initialized       True 
  Ready             True 
  ContainersReady   True 
  PodScheduled      True 
Volumes:
  tls-server-certs:
    Type:        Secret (a volume populated by a Secret)
    SecretName:  function-kcl-tls-server
    Optional:    false
  kube-api-access-4q9sf:
    Type:                    Projected (a volume that contains injected data from multiple sources)
    TokenExpirationSeconds:  3607
    ConfigMapName:           kube-root-ca.crt
    ConfigMapOptional:       <nil>
    DownwardAPI:             true
QoS Class:         BestEffort
Node-Selectors:    nodegroup=crossplane
Tolerations:       dedicated=crossplane:NoSchedule
                   node.kubernetes.io/not-ready:NoExecute op=Exists for 300s
                   node.kubernetes.io/unreachable:NoExecute op=Exists for 300s
Events:
  Type     Reason     Age                  From               Message
  ----     ------     ----                 ----               -------
  Normal   Scheduled  2m32s                default-scheduler  Successfully assigned crossplane/function-kcl-c44536185222-66c9f554fd-m8wht to <node>
  Warning  BackOff    35s                  kubelet            Back-off restarting failed container package-runtime in pod function-kcl-c44536185222-66c9f554fd-m8wht_crossplane(eba0d4c1-b0b0-4c06-b829-0051b7c7590e)
  Normal   Pulled     21s (x3 over 2m31s)  kubelet            Container image "xpkg.upbound.io/crossplane-contrib/function-kcl:v0.7.2" already present on machine
  Normal   Created    21s (x3 over 2m31s)  kubelet            Created container package-runtime
  Normal   Started    21s (x3 over 2m31s)  kubelet            Started container package-runtime
```
</details>


<details>
  <summary>KCL Function Pod - Logs with stacktrace</summary>

```
{"level":"info","ts":1715337555.3369577,"caller":"fn/fn.go:32","msg":"Running Function","tag":""}
panic: fork/exec /usr/local/bin/kclvm_cli: exec format error
goroutine 36 [running]:
 kcl-lang.io/kcl-go/pkg/runtime.(*Runtime).Start(0x400042c8c0)
      /go/pkg/mod/kcl-lang.io/kcl-go@v0.8.6/pkg/runtime/runtime.go:54 +0x1a4
 kcl-lang.io/kcl-go/pkg/runtime.initRuntime(0x0?)
      /go/pkg/mod/kcl-lang.io/kcl-go@v0.8.6/pkg/runtime/init.go:48 +0x14c
 kcl-lang.io/kcl-go/pkg/runtime.GetRuntime.func1()
      /go/pkg/mod/kcl-lang.io/kcl-go@v0.8.6/pkg/runtime/init.go:26 +0x20
 sync.(*Once).doSlow(0x4000bd5530?, 0x0?)
      /usr/local/go/src/sync/once.go:74 +0x100
 sync.(*Once).Do(...)
      /usr/local/go/src/sync/once.go:65
 kcl-lang.io/kcl-go/pkg/runtime.GetRuntime()
      /go/pkg/mod/kcl-lang.io/kcl-go@v0.8.6/pkg/runtime/init.go:26 +0x40
 kcl-lang.io/kcl-go/pkg/service.NewKclvmServiceClient()
      /go/pkg/mod/kcl-lang.io/kcl-go@v0.8.6/pkg/service/client_kclvm_service.go:21 +0x1c
 kcl-lang.io/kcl-go/pkg/kcl.runWithHooks({0x400094ac28, 0x0, 0x0}, {0x64996d0, 0x1, 0x1}, {0x400094ae48, 0x1, 0x1})
      /go/pkg/mod/kcl-lang.io/kcl-go@v0.8.6/pkg/kcl/api.go:477 +0xf4
 kcl-lang.io/kcl-go/pkg/kcl.run(...)
      /go/pkg/mod/kcl-lang.io/kcl-go@v0.8.6/pkg/kcl/api.go:486
 kcl-lang.io/kcl-go/pkg/kcl.RunWithOpts({0x400094ae48?, 0x400094ae48?, 0x0?})
      /go/pkg/mod/kcl-lang.io/kcl-go@v0.8.6/pkg/kcl/api.go:373 +0x50
 kcl-lang.io/kpm/pkg/api.RunWithOpt(0x4000867700)
      /go/pkg/mod/kcl-lang.io/kpm@v0.8.6/pkg/api/kpm_run.go:89 +0x2d0
 kcl-lang.io/cli/pkg/options.(*RunOptions).Run(0x40000da000)
      /go/pkg/mod/kcl-lang.io/cli@v0.8.7/pkg/options/run.go:137 +0x19c
 kcl-lang.io/krm-kcl/pkg/edit.RunKCL({0x4000088788?, 0x1?}, {0x40006c7200?, 0x4000a348c0?}, 0x4000866940)
      /go/pkg/mod/kcl-lang.io/krm-kcl@v0.8.6/pkg/edit/bootstrap.go:52 +0x1ac
 kcl-lang.io/krm-kcl/pkg/edit.(*SimpleTransformer).Transform(0x400094b1d0, {0x4000088788?, 0x1?, 0x0?})
      /go/pkg/mod/kcl-lang.io/krm-kcl@v0.8.6/pkg/edit/transformer.go:46 +0xb0
 kcl-lang.io/krm-kcl/pkg/kio.Filter.Filter({0x400042c000}, {0x4000088788, 0x1, 0x1})
      /go/pkg/mod/kcl-lang.io/krm-kcl@v0.8.6/pkg/kio/filter.go:39 +0x160
 sigs.k8s.io/kustomize/kyaml/kio.Pipeline.ExecuteWithCallback({{0x400094b7b0, 0x1, 0x1}, {0x400094b7a0, 0x1, 0x1}, {0x400094b790, 0x1, 0x1}, 0x0}, ...)
      /go/pkg/mod/sigs.k8s.io/kustomize/kyaml@v0.16.0/kio/kio.go:137 +0x250
 sigs.k8s.io/kustomize/kyaml/kio.Pipeline.Execute({{0x400094b7b0, 0x1, 0x1}, {0x400094b7a0, 0x1, 0x1}, {0x400094b790, 0x1, 0x1}, 0x0})
      /go/pkg/mod/sigs.k8s.io/kustomize/kyaml@v0.16.0/kio/kio.go:104 +0x6c
 main.(*Function).RunFunction(0x40006f57a0, {0x1a59d60?, 0x400011b978?}, 0x40000dc3c0)
      /fn/fn.go:133 +0xe28
 github.com/crossplane/function-sdk-go/proto/v1beta1._FunctionRunnerService_RunFunction_Handler({0x1a59d60, 0x40006f57a0}, {0x255f9f0, 0x400013e1e0}, 0x40007da200, 0x0)
      /go/pkg/mod/github.com/crossplane/function-sdk-go@v0.2.0/proto/v1beta1/run_function_grpc.pb.go:104 +0x1c0
 google.golang.org/grpc.(*Server).processUnaryRPC(0x4000636a00, {0x255f9f0, 0x400013e150}, {0x256ac80, 0x4000737c80}, 0x40005ee900, 0x400084aae0, 0x649c730, 0x0)
      /go/pkg/mod/google.golang.org/grpc@v1.63.2/server.go:1369 +0xb58
 google.golang.org/grpc.(*Server).handleStream(0x4000636a00, {0x256ac80, 0x4000737c80}, 0x40005ee900)
      /go/pkg/mod/google.golang.org/grpc@v1.63.2/server.go:1780 +0xb20
 google.golang.org/grpc.(*Server).serveStreams.func2.1()
      /go/pkg/mod/google.golang.org/grpc@v1.63.2/server.go:1019 +0x8c
 created by google.golang.org/grpc.(*Server).serveStreams.func2 in goroutine 82
      /go/pkg/mod/google.golang.org/grpc@v1.63.2/server.go:1030 +0x13c
```

</details>

### How could this Function help solve your problem?

I don't know whether there's currently a way to make the KCL function work with arm nodes. If not, would it be possible to add support for `arm64` architecture?

---

PS: Excuse me if I picked the wrong labels for the issue, not sure whether you would considered it a bug or it's just unsupported for the moment.

--- a/.github/workflows/ci.yaml
+++ b/.github/workflows/ci.yaml
@@ -33,7 +33,7 @@ env:
   XPKG: xpkg.upbound.io/${{ github.repository}}
 
   # The package version to push. The default is 0.0.0-gitsha.
-  XPKG_VERSION: v0.7.2
+  XPKG_VERSION: v0.7.3
 
 jobs:
   unit-test:
@@ -91,13 +91,13 @@ jobs:
           build-args:
             GO_VERSION=${{ env.GO_VERSION }}
           outputs: type=docker,dest=runtime-${{ matrix.arch }}.tar
-      
+
       - name: Setup the Crossplane CLI
         run: "curl -sL https://raw.githubusercontent.com/crossplane/crossplane/master/install.sh | sh"
 
       - name: Build Package
         run: ./crossplane xpkg build --package-file=${{ matrix.arch }}.xpkg --package-root=package/ --embed-runtime-image-tarball=runtime-${{ matrix.arch }}.tar
-      
+
       - name: Upload Single-Platform Package
         uses: actions/upload-artifact@v3
         with:

--- a/Dockerfile
+++ b/Dockerfile
@@ -36,7 +36,7 @@ RUN --mount=target=. \
 
 # Produce the Function image. We use a very lightweight 'distroless' image that
 # does not include any of the build tools used in previous stages.
-FROM kcllang/kcl AS image
+FROM --platform=${BUILDPLATFORM} kcllang/kcl AS image
 WORKDIR /
 COPY --from=build /function /function
 EXPOSE 9443

--- a/examples/resources/network/functions.yaml
+++ b/examples/resources/network/functions.yaml
@@ -6,4 +6,4 @@ metadata:
     # This tells crossplane beta render to connect to the function locally.
     render.crossplane.io/runtime: Development
 spec:
-  package: xpkg.upbound.io/crossplane-contrib/function-kcl:v0.2.0
+  package: xpkg.upbound.io/crossplane-contrib/function-kcl:latest

